lms/djangoapps/courseware/model_data.py

```diff
@@ -875,6 +877,8 @@ def delete(self, key):
 
         Raises: KeyError if key isn't found in the cache
         """
+        if self.read_only:
```

Comment:
Yeah, it's for completeness. I don't feel great about it, but I thought it would be weird to have a `read_only` flag but still allow deletes.